gms/factories/src/main/java/com/linkedin/gms/factory/common/GraphServiceFactory.java

```diff
@@ -2,27 +2,56 @@
 
 import com.linkedin.metadata.graph.GraphService;
 import com.linkedin.metadata.graph.Neo4jGraphService;
+import com.linkedin.metadata.graph.elastic.ElasticSearchGraphService;
+import com.linkedin.metadata.utils.elasticsearch.IndexConvention;
 import javax.annotation.Nonnull;
+import org.elasticsearch.client.RestHighLevelClient;
 import org.neo4j.driver.Driver;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.DependsOn;
 import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
 
 
 @Configuration
-@Import({Neo4jDriverFactory.class})
+@Import({Neo4jDriverFactory.class, ElasticSearchGraphServiceFactory.class})
 public class GraphServiceFactory {
+  @Autowired
+  @Qualifier("elasticSearchGraphService")
+  private ElasticSearchGraphService _elasticSearchGraphService;
```

Comment:
I think we should do this for neo4j as well. Add a neo4jGraphServiceFactory.

Reply:
+1

Reply:
updated